pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -277,6 +277,64 @@ func (sl portsInOrder) Less(i, j int) bool {
 	return h1 < h2
 }
 
+// EndpointSubsetsEqualIgnoreResourceVersion returns true if EndpointSubsets
+// have equal attributes but excludes ResourceVersion of Pod.
+func EndpointSubsetsEqualIgnoreResourceVersion(subsets1, subsets2 []v1.EndpointSubset) bool {
```

Comment:
I think we should cover this function with an unit test

Reply:
Thanks for the suggestion. Unit test added.

Reply:
What will catch this function being incorrect if any new fields are added to EndpointSubset or any subtypes?

Reply:
> Copying the subsets then resetting resourceVersions fields could but costs more, do you suggest the latter?

not really... that sounds expensive, since you'd have to deep copy first. another possibility:

Use reflection to inspect every field in the structs/substructs, modify each field one at a time in two otherwise equal objects. Make sure the equality check ignores specific fields we want to ignore (like resourceverison), and catches all other differences

Reply:
Thanks for the suggestion. While I was implementating the idea, I found `Equalities` supports adding custom compare function for specific type. If we specify a function for ObjectReference, no extra overhead will be introduced. Actually it will reduce some:
```
BenchmarkEndpointSubsetEquality-40                         64736             16305 ns/op            1674 B/op         69 allocs/op
BenchmarkEndpointSubsetEqualityResourceVersion-40         421993              2920 ns/op             192 B/op         14 allocs/op
BenchmarkEndpointSubsetEqualityResourceVersion2-40         88705             14132 ns/op            1709 B/op         36 allocs/op
```

```
var SemanticIgnoreResourceVersion = conversion.EqualitiesOrDie(
	func(a, b v1.ObjectReference) bool {
		a.ResourceVersion = ""
		b.ResourceVersion = ""
		return a == b
	},
)

func EndpointSubsetsEqualIgnoreResourceVersion2(subsets1, subsets2 []v1.EndpointSubset) bool {
	return SemanticIgnoreResourceVersion.DeepEqual(subsets1, subsets2)
}
```